ofono2mm/mm_bearer.py

```diff
@@ -0,0 +1,139 @@
+        return self.props['IpTimeout'].value
+
+    @dbus_property(access=PropertyAccess.READ)
+    def BearerType(self) -> 'u':
+        return self.props['BearerType'].value
+
+    @dbus_property(access=PropertyAccess.READ)
+    def Properties(self) -> 'a{sv}':
+        return self.props['Properties'].value
+
+    @method()
+    async def Connect(self):
+        await self.doConnect()
+
+    async def doConnect(self):
+        with open('/usr/lib/ofono2mm/ofono_context.xml', 'r') as f:
+            ctx_introspection = f.read()
+        ofono_ctx_object = self.bus.get_proxy_object('org.ofono', self.ofono_ctx, ctx_introspection)
+        ofono_ctx_interface = ofono_ctx_object.get_interface('org.ofono.ConnectionContext')
+        await ofono_ctx_interface.call_set_property("Active", Variant('b', True))
+
+    @method()
+    async def Disconnect(self):
+        await self.doDisconnect()
+
+    async def doDisconnect(self):
+        self.disconnecting = True
+        with open('/usr/lib/ofono2mm/ofono_context.xml', 'r') as f:
+            ctx_introspection = f.read()
+        ofono_ctx_object = self.bus.get_proxy_object('org.ofono', self.ofono_ctx, ctx_introspection)
+        ofono_ctx_interface = ofono_ctx_object.get_interface('org.ofono.ConnectionContext')
+        await ofono_ctx_interface.call_set_property("Active", Variant('b', False))
+
+    async def add_auth_ofono(self, username, password):
+        with open('/usr/lib/ofono2mm/ofono_context.xml', 'r') as f:
+            ctx_introspection = f.read()
+        ofono_ctx_object = self.bus.get_proxy_object('org.ofono', self.ofono_ctx, ctx_introspection)
+        ofono_ctx_interface = ofono_ctx_object.get_interface('org.ofono.ConnectionContext')
+        await ofono_ctx_interface.call_set_property("Username", Variant('s', username))
+        await ofono_ctx_interface.call_set_property("Password", Variant('s', password))
+
+    def ofono_context_changed(self, propname, value):
+        if propname == "Active":
+            if self.disconnecting and (not value.value):
+                self.disconnecting = False
+            elif (not value.value) and self.props['Connected'].value:
+                self.doConnect()
+            self.props['Connected'] = value
+            self.emit_properties_changed({'Connected': value.value})
+        if propname == "Settings":
+            if 'Interface' in value.value:
+                self.props['Interface'] = value.value['Interface']
+                self.emit_properties_changed({'Interface': value.value['Interface'].value})
+                if [value.value['Interface'].value, 2] not in self.mm_modem.props['Ports'].value:
+                    self.mm_modem.props['Ports'].value.append([value.value['Interface'].value, 2])
+            if 'Method' in value.value:
+                if value.value['Method'].value == 'static':
+                    self.props['Ip4Config'].value['method'] = Variant('u', 2)
+                if value.value['Method'].value == 'dhcp':
```

Comment:
Likewise

Reply:
Not a good idea here as both method and interface can be present.